Added an Ansible directory and Getting Started guide for consideration.

jlyman opened the pull request:
Submitting a getting started guide for using Ansible to manage Linode servers. Walks a user (assumed knowledge is low) through installation of Ansible and a basic orientation, and then helps the user set up a very basic LAMP stack on a server using Ansible playbooks.

thetechstewart commented:
@jlyman:  This guide has been assigned to a member of the Linode Technical Writing team for a tech test.  Once we have completed testing, we will follow up with you regarding next steps.

ellekrout commented:
Hello, @jlyman! Just a quick question to make sure I'm reading this correctly -- when you first run the "ansible myserver.com -m ping" code (replacing the myserver.com appropriately, of course), is it supposed to work even *without* a hosts file? When running it, I receive "No hosts match" output on Ubuntu, and an "Unable to find inventory file..." error on OSX.

With a hosts and inventory file, everything runs clean. Should I be getting the successful output even when initially running the command as in First Steps? Just trying to clarify.

Thank you!

jlyman commented:
Ooh, good catch @ellekrout, thank you! (Thought I had run through them all but must have missed this initial one.) 

I've updated that particular section to reflect the proper command, `ansible all -i myserver.com, -m ping`. That should go through without the need for a hosts file.

ellekrout commented:
Thank you for the prompt reply, @jlyman! I'm continuing the tech test for this today, and should get back to you shortly.

ellekrout commented:
Hello once again, @jlyman! I'm completed the tech test, and have created a PR against your branch with some small changes -- mostly just switching "josh" to "username" and other small fixes. I also moved the directory from *Websites* to *Applications* which is where we are transitioning all our configuration management guides.

Two additional comment:
- This guide starts with commands using `sudo`, implying we're working with a non-root user, but I've had to add `sudo` before most `ansible` commands in order to run them as a super user. It's up to you if you would prefer the guide to be run as root or a superuser, so please add or remove `sudo` as needed.
- I did a quick search for this, but passwords must be hashed in the basic user playbook, yes? If so, can you add a note concerning this?

Thank you for you submission! Once we're set with the above things, the guide will be up for copy edit by another member of the team.

jlyman commented:
Thanks @ellekrout. I spotted one additional instance of "josh" that needed to be changed. To reduce confusion, I changed `username` to be `yourusername` in the playbooks to make it stand out as something that required changing so that users wouldn't miss it. I also added an explanation for password hashing.

As for the use of sudo, that surprises me. Were you needing to prefix all ansible and ansible-playbook commands with sudo, on both Ubuntu and OS X? There shouldn't be a need for this. I added the non-root note from the style guide to the top of it, but I'd love to hear more about your experience as it seems contrary to what I'm used to seeing.

ellekrout commented:
The issue was with OSX, I should have noted. Regardless, if you can't replicate the issue, then the added note should be fine. I'll try to replicate it myself later today, or tomorrow.

That said, with the tech review finished, this will be going up for copy edit by another team member shortly!

alexfornuto commented on docs/applications/ansible/getting-started-with-ansible.md:
```diff
@@ -0,0 +1,357 @@
+
+This guide will assume a brand new Ubuntu 14.04 LTS server, without any additional configuration already done to the box. The very first order of business will be to add in our public encryption keys so that we can connect without supplying passwords.
+
+{: .note}
```
I would consider removing this note entirely. Since Ansible will give you the same prompt as SSH there' no difference, and I think we can assume that someone setting up Ansible is already familiar with the RSA fingerprint warning.

jlyman replied:
Good call, I removed it.

alexfornuto commented:
Is there any common convention for where playbooks should be stored?

alexfornuto commented:
I've created a PR with my copy edit: https://github.com/jlyman/docs/pull/2. I assume it can't be auto merged because of your last commit. I ended up deleting my comment and just making the change I commented on. You may find it easiest to roll back your branch by one commit in order to merge my copy edit.

Please let me know if you have any questions or comments about my copy edit before you merge it.

jlyman commented:
Thanks @alexfornuto. I resolved the conflict and merged in the PR. I also changed one word (monotany -> monotony). 

In regards to your question about where playbooks should be stored, there is not that I've ever come across. (Some people store them with their sites, others with a grouping of servers, etc., really depends). There are suggestions on how to arrange more complicated playbooks, but I felt it out of scope for the intro guide.

Thanks for the copy edit, looked great!

alexfornuto commented:
@jlyman before we can publish, please let us know if you would like your bounty applied as credit to a Linode account or over PayPal. In either case we need the email address associated.

jlyman commented:
@alexfornuto Awesome! Let's do PayPal: joshua@sparkgenius.com

Anything else we need to wrap this one up?

alexfornuto commented:
@jlyman We're just going to do one last review of the metadata and keywords, and then we'll publish it! I should get back to you by EOD today.

alexfornuto commented:
Hello @jlyman 

Your article is now live! It can be found at https://linode.com/docs/applications/ansible/getting-started-with-ansible. Your bounty payment should be processed within the day.

Thank you for contributing to Linode Guides & Tutorials!

All the best,
@alexfornuto
 The Linode Training & Education Team

jlyman commented:
Fantastic! Thanks to everyone involved.
